client/my-sites/email/email-management/home/email-plan/index.jsx

```diff
@@ -340,33 +347,38 @@ function EmailPlan( { domain, hideHeaderCake = false, selectedSite, source } ) {
 			<DocumentHead title={ titleCase( getHeaderText() ) } />
 			<MailPoetUpsell />
 			{ ! hideHeaderCake && <HeaderCake onClick={ handleBack }>{ getHeaderText() }</HeaderCake> }
-			<EmailPlanHeader
-				domain={ domain }
-				hasEmailSubscription={ hasSubscription }
-				isLoadingEmails={ isLoading }
-				isLoadingPurchase={ isLoadingPurchase }
-				purchase={ purchase }
-				selectedSite={ selectedSite }
-				emailAccount={ getAccount( emailAccounts ) }
-			/>
+			{ ! hideHeader && (
+				<EmailPlanHeader
+					domain={ domain }
+					hasEmailSubscription={ hasSubscription }
+					isLoadingEmails={ isLoading }
+					isLoadingPurchase={ isLoadingPurchase }
+					purchase={ purchase }
+					selectedSite={ selectedSite }
+					emailAccount={ getAccount( emailAccounts ) }
+				/>
+			) }
 			<EmailPlanMailboxesList
 				account={ getAccount( emailAccounts ) }
 				domain={ domain }
 				mailboxes={ getMailboxes( emailAccounts ) }
 				isLoadingEmails={ isLoading }
+				addMailboxPath={ hidePlanActions && getAddMailboxProps()?.path }
 			/>
-			<div className="email-plan__actions">
-				<VerticalNav>
-					{ renderAddNewMailboxesOrRenewNavItem( getMailboxes( emailAccounts ) ) }
-					<UpgradeNavItem
-						currentRoute={ currentRoute }
-						domain={ domain }
-						selectedSiteSlug={ selectedSite.slug }
-					/>
-					{ renderManageAllMailboxesNavItem() }
-					{ renderViewBillingAndPaymentSettingsNavItem() }
-				</VerticalNav>
-			</div>
+			{ ! hidePlanActions && (
+				<div className="email-plan__actions">
+					<VerticalNav>
+						{ renderAddNewMailboxesOrRenewNavItem( getMailboxes( emailAccounts ) ) }
+						<UpgradeNavItem
+							currentRoute={ currentRoute }
+							domain={ domain }
+							selectedSiteSlug={ selectedSite.slug }
+						/>
+						{ renderManageAllMailboxesNavItem() }
+						{ renderViewBillingAndPaymentSettingsNavItem() }
+					</VerticalNav>
+				</div>
+			) }
```

Comment:
Thanks for the ping! Can you clarify which links you're talking about? Is it for managing payments?

Reply:
Sorry, I didn't attach the screenshot in the previous comment. Yes, one of the links is "View billing and payment settings".

<img width="1114" alt="394249786-ad1e2ae1-dbfb-46b2-97c2-f60edf28812b" src="https://github.com/user-attachments/assets/aa90a846-9376-4538-b87b-542a2a087a4e">

Reply:
Cool, thank you for sharing that. It must have been a oversite on my part. Can we add it to the more menu dropdown like this:

![image](https://github.com/user-attachments/assets/8ea4eec6-09ad-4dde-b983-541f129268cf)

Reply:
I guess @bogiii already addressed this [here](https://github.com/Automattic/wp-calypso/pull/97252#issuecomment-2532298156), saying that this is planned for a separate PR.

A quick though from me is that it might makes sense to refactor `EmailPlan` to make it more modular, as part of that effort.

@fditrapani, do we intend to keep the existing email management pages in Calypso the way they currently are?

Reply:
Any bigger refactor proposed in this review, such as transforming the files to the `.ts` of this PR, would be nice to be handled in a separate PR for a more straightforward review.

It is worth mentioning that there will be a series of PRs related to this domain management revamp, so I would avoid handling everything in one PR.

> do we intend to keep the existing email management pages in Calypso the way they currently are

Yeah, the plan is to keep both versions for some time, and then, in one moment, we can keep only one (new, revamped) version.